cmd/fleetctl/preview.go

```diff
@@ -630,34 +630,40 @@ func downloadOrbitAndStart(destDir, enrollSecret, address, orbitChannel, osquery
 	}
 
 	updateOpt := update.DefaultOptions
-	switch runtime.GOOS {
-	case "linux":
-		updateOpt.Platform = "linux"
-	case "darwin":
-		updateOpt.Platform = "macos"
-	case "windows":
-		updateOpt.Platform = "windows"
-	default:
-		return fmt.Errorf("unsupported arch: %s", runtime.GOOS)
+
+	if runtime.GOOS == "darwin" {
+		// We need to initialize updates for latest orbit which does not
+		// support .app bundle yet.
+		updateOpt.Targets = update.DarwinLegacyTargets
```

Comment:
@mna (Answering your question about why we need to keep support for legacy darwin `osqueryd` target.)

`fleetctl preview` fetches `orbit` and `osqueryd` from the TUF server. We need to support legacy to not break `fleetctl preview` until the `.app` version of `osqueryd` and the `orbit` version that support this new functionality are available in the TUF server.

So, on next release after this one (probably `4.13.0`) we can get rid of `update.DarwinLegacyTargets`.

`update.DarwinLegacyTargets` is also used in `orbit/cmd/orbit/orbit.go` to allow running `orbit` in "development mode", which also pulls `osqueryd` by default from the TUF server.